letsencrypt-auto-source/pieces/pipstrap.py

```diff
@@ -123,7 +125,24 @@ def read_chunks(response, chunk_size):
     return path
 
 
+def pypi_available():
+    """
+    Check if pypi is being blocked via DNS
+    :return: bool
+    """
+    try:
+        return bool(socket.gethostbyname('pypi.python.org'))
+    except socket.error:
+        return False
+
+
 def main():
+    if not pypi_available():
+        print("Unable to resolve DNS name pypi.python.org. You may be experiencing internet connectivity problems, or "
+              "it may be blocked due to regional filtering. If you suspect that to be the case, you may need to set a "
+              "pip mirror in ~/.pip/pip.conf. See GitHub issue https://github.com/certbot/certbot/issues/2516")
```

Comment:
I think it might be nice if this referenced some sort of documentation URL instead of just a GitHub issue. I think that would better allow for evolving content over time and providing clear, concise information to users who receive this error message.

The over-engineering-inclined part of my brain likes the idea of including the pipstrap version as a part of the URL (perhaps as a fragment or query parameter). For now it could be unused, but including it now allows for easy customization of the help page based on version if that's ever necessary (e.g., if we ever decide to improve the user experience here).

In line with @bmw's comment on the associated issue (https://github.com/certbot/certbot/issues/2516#issuecomment-303238072), I think the it might be best if the documentation were associated with pipstrap itself and not Certbot, since that project is the best place provide this documentation. Perhaps it could be added to the pipstrap repo as a `gh-page`. (Although I'd defer to @erikrose on that.)